app/jobs/repo_setup_job.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require "concerns/repo_setup.rb"
+
+class RepoSetupJob < ApplicationJob
+  queue_as :repo_setup
+
+  def perform(assignment_repo_type, assignment_repo_id)
+    @assignment_repo = get_assignment_repo assignment_repo_type, assignment_repo_id
+
+    return unless @assignment_repo
+
+    if @assignment_repo.github_repository.import_progress[:status] != "complete"
+      RepoSetupJob.set(wait: 1.hour).perform_later(assignment_repo_type, assignment_repo_id)
```

Comment:
This is a possibility, I am going to hold out on this PR for now and push out another PR with importer things.